
syntax = "proto3";

import "google/protobuf/empty.proto";

package Exercise;

message File 
{
	string filename = 1;
	bytes chunk = 2;
}

message Parameters 
{
	int64 number = 1;
	string text = 2;
}

service DataExchange 
{
	rpc GetParameters(google.protobuf.Empty) returns (Parameters) {}
	rpc SetParameters(Parameters) returns(google.protobuf.Empty) {}
	rpc GetFile(File) returns (stream File) {}
	rpc SetFile(stream File) returns (google.protobuf.Empty) {}
}
